### ISSUE TITLE
Thanks and request for extension - app_protocol 

### ISSUE THREAD

**Issue by A-n-d-y-K** [compose-viz/compose-viz#54]
Dear @wst24365888,
1st of all, many thanks for this public repo. 

I have forked it and added [app_protocol](https://docs.docker.com/compose/compose-file/05-services/#long-syntax-3) and a legend option to your current tag v0.3.1.

To be honest I´m not a python expert but I´m willing to learn.

With this issue I wanted to ask you if you are interested in adding this to your repo. I sure there what I did is many not good enough. E.g. legend is hard coded, no tests are implemented.

Nevertheless I kindly ask for your guidance and the steps needed for a v0.3.2 ;-)

Find my changes there [add feature --legend, yml attr "app_protocol"](https://github.com/A-n-d-y-K/compose-viz/commit/0aa9bab07908e2804120e16254148e50f19d2bab)

A rendered svg from your example [non-normative-with-legend.VizFormats.svg](https://github.com/A-n-d-y-K/compose-viz/commit/0aa9bab07908e2804120e16254148e50f19d2bab#diff-0da93bda3a4439523cabfeaeec680b5d765385a5c13982c085e8d9822530263d)

Best regards

**Comment by A-n-d-y-K** [comment]
Dear @wst24365888 ,
is there a chance to get feedback from your side?
Thanks much in advance

**Comment by wst24365888** [comment]
Hi @A-n-d-y-K , thank you for bringing up this issue, and sorry for the late reply. Our team has been quite busy, but we will discuss the issues and the new version over this weekend.

**Comment by wst24365888** [comment]
Hey @A-n-d-y-K , we've reviewed your issue (I've left some comments) and the idea of `legend` and adding `app_protocol` is great!

For the `legend` part, we decided to change the legend layout from vertical to horizontal.
For the `app_protocol` part, which was introduced in the recent version of `compose-spec`, we think that  continouosly modifying `compose_spec.py` is not the best way to do it, so we've adopted a new cron job to update the submodule (the `compose-spec` repo) and generate `compose_spec.py` according to the new schema.

Again, thanks for bringing up this issue!
